boxoffice/static/js/views/invoice.js

```diff
@@ -0,0 +1,127 @@
+window.Boxoffice = window.Boxoffice || {};
+
+window.Boxoffice.Invoice = {
+  config: {
+    baseURL: window.location.origin,
+    submit: {
+      method: 'POST',
+      urlFor: function(id) {
+        return Boxoffice.Invoice.config.baseURL + '/order/' + id + '/invoice';
+      }
+    }
+  },
+  init: function(invoiceDetails) {
+    var invoice = this;
+    invoice.formComponent = new Ractive({
+      el: '#boxoffice-invoice',
+      template: '#boxoffice-invoice-form-template',
+      data: {
+        invoiceDetails: invoiceDetails,
+        editDetails: true,
+        errorMsg: ""
+      },
+      scrollTop: function() {
+        $('html,body').animate({ scrollTop: $(invoiceComponent.el).offset().top }, '300');
+      },
+      submitInvoiceDetails: function(event) {
+        var validationConfig = [{
+          name: 'taxid',
+          rules: 'max_length[255]'
+        },
+        {
+          name: 'invoicee_name',
+          rules: 'required|max_length[255]'
+        },
+        {
+          name: 'street_address',
+          rules: 'required|max_length[255]'
+        },
+        {
+          name: 'city',
+          rules: 'required|max_length[255]'
+        },
+        {
+          name: 'state',
+          rules: 'required|max_length[255]'
+        },
+        {
+          name: 'country',
+          rules: 'required|max_length[255]'
+        },
+        {
+          name: 'postcode',
+          rules: 'required|max_length[255]'
```

Comment:
Let's remove the `max_length` checks and delegate this check to the server? `required` is useful since it gives the user immediate feedback. The problem with `max_length` is it now needs to be in sync with the backend's `max_length` as well.